listener.go

```diff
@@ -79,14 +79,12 @@ func (l *listener) handleConn(conn net.Conn) {
 	l.mx.Unlock()
 
 	defer func() {
+		session.Close()
+		conn.Close()
 		l.mx.Lock()
-		closed := l.closed
+		delete(l.sessions, sessionID)
```

Comment:
This line certainly seems important @oxtoacart. Is our production code running with this?